src/server/service/kv.rs

```diff
@@ -625,20 +625,20 @@ impl<T: RaftStoreRouter<RocksEngine> + 'static, E: Engine, L: LockManager> Tikv
                 RAFT_MESSAGE_RECV_COUNTER.inc();
                 let to_store_id = msg.get_to_peer().get_store_id();
                 if to_store_id != store_id {
-                    future::err(Error::from(RaftStoreError::StoreNotMatch {
+                    warn!("store not match"; "self" => store_id, "given" => to_store_id);
+                    return future::err(Error::from(RaftStoreError::StoreNotMatch {
                         to_store_id,
                         my_store_id: store_id,
-                    }))
-                } else {
-                    let ret = ch.send_raft_msg(msg).map_err(Error::from);
-                    future::ready(ret)
+                    }));
+                } else if let Err(e) = ch.send_raft_msg(msg) {
+                    warn!("dispatch raft msg from gRPC to raftstore fail"; "err" => ?e);
```

Comment:
I'm afraid it can produce too many logs. And the flow control side effect is removed by this way, how to stop remote from sending a lot of messages when downstream is overloaded?